### ISSUE TITLE
`RUF031` (`incorrectly-parenthesized-tuple-in-subscript`) - false positive on implicit type alias

### ISSUE THREAD

**Issue by Garrett-R** [astral-sh/ruff#15264]
EDIT: Ah, I see that this issue is known and described in bullet point 3 of [this comment](https://github.com/astral-sh/ruff/issues/13065#issuecomment-2306282918).  Decided to keep this issue since duplicates can help people find the original issue.

_______________________

## Description

With setting the config: `"parenthesize-tuple-in-subscript": true`, I get this:

```python
type Foo = dict[str, str]
Foo = dict[str, str]  # (RUF031:Use parentheses for tuples in subscripts)
```
([playground](https://play.ruff.rs/4df86970-fc1e-408e-aa37-05324792c217))
